Add tests for shouldRetry on parents of closed linked cursors

chrisvest opened the pull request:
This is a scenario that shows up in the GBPTree usage, but wasn't yet covered by the page cache tests. Well, it is now.

chrisvest committed:
Add tests for shouldRetry on parents of closed linked cursors

burqen commented on community/io/src/test/java/org/neo4j/io/pagecache/PageCacheTest.java:
```diff
@@ -2796,6 +2796,65 @@ public void shouldRetryFromUnboundLinkedWriteCursorMustNotThrow() throws Excepti
+              PageCursor cursor = pf.io( 0, PF_SHARED_WRITE_LOCK ) )
+        {
+            assertTrue( cursor.next() );
+            //noinspection unused
```
Does not need `//noinspection unused`

burqen commented on community/io/src/test/java/org/neo4j/io/pagecache/PageCacheTest.java:
```diff
@@ -2796,6 +2796,65 @@ public void shouldRetryFromUnboundLinkedWriteCursorMustNotThrow() throws Excepti
+            {
+                assertTrue( linked.next() );
+            }
+            cursor.shouldRetry();
```
Why not `assertFalse( cursor.shouldRetry() )`?

chrisvest replied:
Because we _could_ race with eviction here.

chrisvest replied:
So the `assertFalse` should be removed from the other test, actually.

burqen reviewed (changes requested):
Some small comments

chrisvest committed:
Small page cache test cleanup

Including removing an assertFalse on shouldRetry, that had a non-zero chance of racing with eviction, which _in principle_ makes it flaky.

chrisvest commented:
@burqen Addressed your comments.

burqen commented:
@chrisvest Sweet! Will wait until it goes green.
